### ISSUE TITLE
Various Ahnonay Graphical Issues.

### ISSUE THREAD

**Issue by Traveler263** [H-uru/moul-assets#115]
MOUL's version of ahnonay has various Graphical issues that it's CC counterpart does not have.

These include:

Sphere 4 is too dark to safely navigate in in many places.
Water in sphere 1 is stuck visually in "currents on always" mode- even when currents are turned off.
Sphere 1 missing distant fog/mist.
Sphere 2 has bright textures on the rocks under the fog layer
The maintenance rooms all have z Fighting on the walls (Some more prominently than others, to the point of eye strain)
Quabs seem to have a lower resolution texture too (I may be mis-remembering this one, though, feel free to ignore)
The Tower in Sphere 1 has a circular spotlight pointed at an object that no longer exists
And the outside waterfall animation is visibly resetting when it loops

**Comment by Mystler** [comment]
Sphere 2 issue of missing vertex colors was fixed in #296.

I have locally converted the Quabs from PotS but just initial discussion has already shown that replacing the quab model (or doing more advanced shenanignas) is apparently very subjective which makes it not just a graphical issue.

A lot of the other issues, i cannot really reproduce so they may have gotten fixed before. I'd suggest closing this generic issue and opening specific issues for specific points instead, if they still apply.

**Comment by dpogue** [comment]
> Sphere 1 missing distant fog/mist.

In PotS there are 5 different fog values provided (one is commented out, but the comment says where it was intended to be used), and in MOUL it's all 1 Age so there's only 1 set of fog values.

This _is_ fixable by adding plFogEnvironments to every drawable based on what sphere it's in, but that is likely to bloat the PRP files quite a bit (`sizeof(plFogEnvironment) + sizeof(plKey) * number_of_icicles`).

If we had [fog messages](https://github.com/H-uru/Plasma/issues/711) we could do region-based triggers with responders, but realistically the solution right now is probably to have regions that trigger a Python script that changes the global fog values (but only on the local client).

**Comment by Mystler** [comment]
I forgot if python can run console commands on external clients. A cheap version might be to just set fog via console commands based on what sphere gets loaded in the init function, with a special fifth case in the sphere 4 maintenance room.

**Comment by Mystler** [comment]
I also noticed the spotlight in the tower, which looks like it's baked into vertex colors, so a lot of effort to fix for the return.
Either way, i think i'd really like separate issues rather than this catch all, because it's difficult to track progress on it (or potentially even ever get it done).

**Comment by Hazado** [comment]
> I forgot if python can run console commands on external clients. A cheap version might be to just set fog via console commands based on what sphere gets loaded in the init function, with a special fifth case in the sphere 4 maintenance room.

https://github.com/H-uru/Plasma/blob/master/Scripts/Python/xFogSet.py

**Comment by Mystler** [comment]
Yeah, this is working pretty great, I have it in purely via python locally now.

MOUL:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/bca0a199-9eb9-43a0-9523-17c983ac44f1" />

With TPotS fog:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/4f54b00f-35bd-4063-a9d8-bae8b36d2173" />

**Comment by Mystler** [comment]
Sphere 2 Before/After:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/0897328e-0503-4ef7-bc59-779633b03c05" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/80671df9-bf29-4936-b1a7-43cb0b2da7e8" />

Sphere 3 is unchanged because that's the default values in the MOUL fni.

Sphere 4, we have a problem. For this one, i'd suggest not using the TPotS one and also staying on the default because:

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ff933fe7-5736-4180-aa52-04a8e02a7cf6" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/0430c333-505c-4f98-aa89-8c82175b5487" />

**Comment by EhrenCG** [comment]
What's going on with your pictures?  Are you not trying to copy all the fni settings?

This is MOUL with TPotS fog (you seem to be missing the SetDefColor .7 .7 .7 fog color):

<img width="1600" height="1200" alt="Image" src="https://github.com/user-attachments/assets/4e27d3ba-cb47-49e3-9cf8-72a1e6feba22" />

This is also MOUL with the corresponding TPotS fog  (you seem to have missed the fog color of SetDefColor 1 1 1 and SetYon 15000):

<img width="1600" height="1200" alt="Image" src="https://github.com/user-attachments/assets/aeb18777-2db5-4189-b47a-81a9cd722fab" />

**Comment by Mystler** [comment]
I set those exact values (except the SetYon because that is not exposed to python but shouldn't really make a big visual difference afaict). I'm confused if the python api is doing something silly here.

**Comment by Mystler** [comment]
Ahh, i found the mistake.

**Comment by Mystler** [comment]
Fixed with new screenshots over in https://github.com/H-uru/Plasma/pull/1942

**Comment by Mystler** [comment]
As a side note: I have put my TPotS quab conversion (changed CustomAvatars file) up on a branch at https://github.com/Mystler/moul-assets/tree/quabs-tpots. Given that this seems to be subjective i have no intention to open a PR unless I get an overwhelming response that this replacement is wanted, but I want the file to be available for everyone who cares.
